speech/google/cloud/speech/streaming_response.py

```diff
@@ -0,0 +1,152 @@
+                 result_index=None):
+        results = results or []
+        self._error = error
+        self._endpointer_type = EndpointerType.REVERSE_MAP.get(
+            endpointer_type, None)
+        self._result_index = result_index
+        self._results = [StreamingSpeechResult.from_pb(result)
+                         for result in results]
+
+    @classmethod
+    def from_pb(cls, pb_response):
+        """Factory: construct a ``StreamingSpeechResponse`` from protobuf.
+
+        :type pb_response:
+            :class:`google.cloud.speech.v1beta1.StreamingRecognizeResponse`
+        :param pb_response: Instance of protobuf
+                            ``StreamingRecognizeResponse``.
+        :rtype: :class:`~StreamingSpeechResponse`
+        :returns: Instance of ``StreamingSpeechResponse``.
+        """
+        error = pb_response.error
+        endpointer_type = pb_response.endpointer_type
+        results = pb_response.results
+        result_index = pb_response.result_index
+        return cls(error=error, endpointer_type=endpointer_type,
+                   results=results, result_index=result_index)
+
+    @property
+    def confidence(self):
+        """Confidence score for recognized speech.
+
+        :rtype: float
+        :returns: Confidence score of recognized speech [0.0-1.0].
+        """
+        if self.results and self.results[0].alternatives:
+            top_alternative = self.results[0].alternatives[0]
+            return top_alternative.confidence
+        else:
+            return 0.0
+
+    @property
+    def endpointer_type(self):
+        """Endpointer indicating the state of the speech detection.
+
+        :rtype: str
+        :returns: String derived from :class:`~endpointer_type.EndpointerType`.
+        """
+        return self._endpointer_type
+
+    @property
+    def is_final(self):
+        """Represents an interim result that may change.
+
+        :rtype: bool
+        :returns: True if the result has completed it's processing.
+        """
+        if self.results:
+            return self.results[0].is_final
+        else:
+            return False
+
+    @property
+    def result_index(self):
+        """Result index associated with this response.
+
+        :rtype: int
+        :returns: Result index of this response.
+        """
+        return self._result_index
+
+    @property
+    def results(self):
+        """List of results for this response.
+
+        :rtype: list of :class:`~result.StreamingSpeechResult`
+        :returns: List of ``StreamingSpeechResult`` in this response.
+        """
+        return self._results
+
+    @property
+    def transcript(self):
+        """Get most likely transcript from response.
+
+        :rtype: str
+        :returns: Transcript text from response.
+        """
+        if self.results and self.results[0].alternatives:
+            top_alternative = self.results[0].alternatives[0]
+            return top_alternative.transcript
```

Comment:
Thanks for this change, though the variable doesn't make me feel any better about what is happening.

Can you

1. Expalin why we only care about `results[0]` and `alternatives[0]`?
1. If the answer is "there should only be one", then `self.results` should become `self.result` (or an evan better name than `result`) (Similar if there should only be one `alternatives`)
1. If we expect there to only be one when this is accessed (but not always), then `self.result` won't be possible. But in that case you should enforce the constraint on the caller (i.e. make sure `len(self.results) == 1` and similar for `alternatives`)
1. Whatever the conclusion is, can you add a comment to the code explaining why?